cmd/botkube-agent/main.go

```diff
@@ -61,6 +61,8 @@ const (
 	reportHeartbeatMaxRetries = 30
 )
 
+var healthNotifiers = make(map[string]health.Notifier)
```

Comment:
1. it doesn't make sens to have a global var here. Let's more it to the  healtChecker that is responsible for it
2. no need to repeat the interfaces (`HealthNotifierSink`/`HealthNotifierBot`), keep it in a single place (healthChecker)
2. if / else statement is repeated, while it's common for all bots, please see linked code changes
3. FailedSink and FailedBot, referring to the same for the healthChecker - it doesn't care, just have single one
4. You don't need to be worried about integration name, you can use it on 'nil'
    ```go
     func Test(t *testing.T) {
 	    app, err := NewSocketSlackDemo()
 	    assert.Error(t, err)
 	    
 	    fmt.Println(app.IntegrationName())
     }
     
     func NewSocketSlackDemo() (*SocketSlack, error) {
 	    return nil, errors.New("fake")
     }
    ```


Changes in code based on feedback above: https://github.com/kubeshop/botkube/commit/c472d4d285c8208b099bac588bbcad780ec82fdb